### ISSUE TITLE
Fields for media attribution (e.g. credits/author/copyright/source/license) and option for admins to make them required

### ISSUE THREAD

**Issue by xtaran** [mastodon/mastodon#22931]
### Pitch

It would be great if 

1. posting pictures and videos would allow to include a dedicated field stating the media's author and maybe also license, and
2. that instance administrators can optionally require their users to fill in these fields.

I imagine a workflow like this:

* If posting happens inside an app and a picture is taken with the app, the picture is by default marked as own work and maybe optionally filled with the preferred license of the poster. (This is something which is likely only relevant for mobile device apps, but an API to enable this is necessary then.)
* Otherwise (e.g. in the web app) posting a picture or video not only allows a textual description as of now, but also fields for author/copyright/source (same or separate fields, not sure) and a license (clearly a separate field).

Some more thoughts:

* It would be nice if there's a set of known licenses to easily choose from. I'd suggest to use license shortnames as used on https://spdx.org/licenses/, at least for FLOSS licenses.

Related issues:

* #22534 — I'd say this is part of what I imagine, but not all of it.
* #9528 — Yes, but that's just for the header picture, i.e. the motivation is to allow users to state there is a license and copyright. My motivation (see below) is more the other way round, not to enable users but to encourage (or optionally enforce) users to state author and license of media.

### Motivation

The idea came up when I saw people discussing accounts like https://mastodon.world/@Kashalot or https://mstdn.social/@Lost_World which are very likely posting mostly media by others without attribution or credits.

So far, these types of accounts seem still seldom in the Fediverse, but are quite widespread on Twitter and I fear that there will come more of them to the Fediverse in the future.

IMHO such accounts pose a potential legal thread to instance administrators. So giving administrators the option to enforce stating media credits/authors/copyright and licenses might give them additional legal safety against users which wildly copy media files from anywhere and posting them without any credits.

**Comment by Ciantic** [comment]
I find ALT-text policing in Mastodon fascinating. It could work for attribution as well!

Imagine if you are uploading an image or video, you should choose:

```
[  ] I made this content
[X] I found it elsewhere, type here an attribution: _______

```

I don't know how complicated the fields should be, but simple could work.

This way we would store the metadata if the content originated from the user, or came from elsewhere.

It would allow the filtering of original content, and help discovering makers and creators easier.

**Comment by skopf** [comment]
I agree that there is a potential legal thread for instance admins and that the situation is likely getting worse in the future.

What makes this situation more complicated is the aspect of federation. When a user provides content to the home instance, then this content will be replicated to other instances. In the case of a public toot for example, it will be displayed on the websites of these other instances as part of their federated timelines.

If a user would use this new feature to set a restrictive license like CC-BY-NC, then commercial instances like mastodon.social would not be allowed to display this content. Probably not even the own home instance.

Introducing this new feature would need to be combined with an update to the instance TOS that clearly states that the home instance is licensed to use this content in addition to whatever license the user selects.
Also the visibility option of the toot needs to be linked to this license selection to guarantee that the instance complies with the selected license. It can only federates toots where allowed by the license (e.g. public toots would only be possible with CC0 or CC-BY).

**Comment by SleepyBrett** [comment]
I don't see why this should be limited to posts with media attached. Text also falls under copyright.
